File: python/v1.4/get_account_tree.py
#!/usr/bin/python
#
# Copyright 2014 Google Inc. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

"""This example gets a specific account for the logged in user.

This includes the full tree of sub-accounts.

Tags: accounts.get
"""

__author__ = 'jalc@google.com (Jose Alcerreca)'

import sys
from apiclient import sample_tools
from oauth2client import client
from adsense_util import get_account_id


def main(argv):
  # Authenticate and construct service.
  service, _ = sample_tools.init(
      argv, 'adsense', 'v1.4', __doc__, __file__, parents=[],
      scope='https://www.googleapis.com/auth/adsense.readonly')

  try:
    account_id = get_account_id(service)
    # Retrieve account.
    request = service.accounts().get(accountId=account_id, tree=True)
    account = request.execute()

    if account:
      display_tree(account)

  except client.AccessTokenRefreshError:
    print ('The credentials have been revoked or expired, please re-run the '
           'application to re-authorize')


def display_tree(account, level=0):
  print (' ' * level * 2 +
         'Account with ID "%s" and name "%s" was found. ' %
         (account['id'], account['name']))

  if 'subAccounts' in account:
    for sub_account in account['subAccounts']:
      display_tree(sub_account, level + 1)

if __name__ == '__main__':
  main(sys.argv)

File: v2/php/examples/GetAccountTree.php
<?php
/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Gets a specific account for the logged in user.
 * This includes the full tree of sub-accounts.
 *
 * Tags: accounts.get
 */
class GetAccountTree {
  /**
   * Gets a specific account for the logged in user.
   *
   * @param $service Google_Service_AdSense AdSense service object on which to
   *     run the requests.
   * @param $accountId string the ID for the account to be used.
   */
  public static function run($service, $accountId) {
    $separator = str_repeat('=', 80) . "\n";
    print $separator;
    printf("Displaying AdSense account tree for %s\n", $accountId);
    print $separator;

    $account = $service->accounts->get($accountId);
    self::displayTree($service, $account, 0);

    print "\n";
  }

  /**
   * Auxiliary method to recurse through the account tree, displaying it.
   */
  private static function displayTree($service, $parentAccount, $level) {
    print str_repeat(' ', $level);
    printf("Account with ID \"%s\" and name \"%s\" was found.\n",
        $parentAccount['name'], $parentAccount['displayName']);

    $childAccounts = $service->accounts->listChildAccounts($parentAccount['name']);
    if (!empty($childAccounts)) {
      foreach ($childAccounts as $childAccount) {
        self::displayTree($childAccount, $level + 1);
      }
    }
  }
}
